index.js

```diff
@@ -47,6 +62,9 @@ function pipeLogsToTerminal(config) {
           ' ' +
           options.consoleProps.URL;
       }
+      if (options.name === 'request') {
+        options.message = options.renderProps.message.replace(/---\ /, '');
```

Comment:
Add the log here first. This is used when the `request` fails.

Reply:
So just that I understand: We are adding this log that actually signifies that request has started and then in the command:ended we actually add the response details right?

Reply:
If yes could we maybe move this also in the command:ended and here in the log:added we just skip the logging of the request command? Just so that we have in one place the handling of the cy.request command output.

Reply:
At first I did what you said. Skip the request log here. Logged the request command at `Cypress.on('command:end'` However, this implementation does not log when the request fails. To log failed requests we need to log at the beginning of the request and log the response at when the request completes. Of course if the request fails there will be no response to log.

Reply:
What do you mean by fail? We get a non 2xx status code? Or timeout? Because 400 and 500 errors might actually contain important response information and we should log those too.

Reply:
I mean none `2xx` status code and timeout. 

`Cypress.on('command:end', async ({attributes}) => {` will only return `2xx` responses. When a response `4xx` or `5xx` is returned like the screenshot below, `command:end` will not be reached. 

This is why I didn't log `request` at `command:end` 

![Screen Shot 2020-02-25 at 10 05 40 am](https://user-images.githubusercontent.com/5380757/75199153-65b13c80-57b6-11ea-9f10-3ab442d820b1.png)

Reply:
Thats quite unfortunate. You have mentioned that we have access to the unresolved promise in the log added event. We could try using that, and most likely in that case we d handle all responses.

Reply:
But I don't know how to do that. `options.consoleProps` contains the response. 
![Screen Shot 2020-02-25 at 10 19 13 am](https://user-images.githubusercontent.com/5380757/75199849-4c10f480-57b8-11ea-8a9d-a8c6b3b5d4c6.png)

But it is only populated after the request is resolved. How do you wait for the request to resolve?

Reply:
Did you check for any promises in the options? If there is none we could try the command start event, if that has nothing of use either what we cand do is wrap (extend) the cy.request command and there we will have access ( like the way the route command is wrapped )

Reply:
I will try to wrap `request`

Reply:
```
  Cypress.Commands.overwrite('request', async (originalFn, options = {}) => {
    try {
      const a = await originalFn(options);
      console.log('-->', options, a);
      return a;
    } catch (e) {
      console.log('error ', JSON.stringify(e), e);
      throw e;
    }
  });
```

It is cleaner to write it this way. The none `2xx` response are thrown as a `Cypress.error`. Do you want to go down this path?